walkeeper/src/replication.rs

```diff
@@ -216,16 +216,18 @@ impl ReplicationConn {
 
             // Read some data from the file.
             let mut file_buf = vec![0u8; send_size];
-            file.seek(SeekFrom::Start(xlogoff as u64))?;
-            file.read_exact(&mut file_buf)?;
+            file.seek(SeekFrom::Start(xlogoff as u64))
+                .and_then(|_| file.read_exact(&mut file_buf))
+                .context("Failed to read data from WAL file")?;
 
             // Write some data to the network socket.
             pgb.write_message(&BeMessage::XLogData(XLogDataBody {
                 wal_start: start_pos.0,
                 wal_end: end_pos.0,
                 timestamp: get_current_timestamp(),
                 data: &file_buf,
-            }))?;
+            }))
+            .context("Failed to send XLogData")?;
```

Comment:
These two errors are quite different: The first indicates that something is pretty badly wrong: the point of a safekeeper is to not lose the WAL, so if it it cannot read the WAL file, that's serious. In contrast, the second error can happen whenever the network connection is broken, which is more or less normal.

I think this is OK for now, a message is nicer than no message, which is what we had before this PR. But this is a good example to consider in https://github.com/zenithdb/zenith/issues/551.

Reply:
After thinking about this for a bit: is there any reason here that we shouldn't just panic, if things are clearly pretty bad?

And if we shouldn't panic, what would the optimal behavior be?